packages/expo-updates/ios/EXUpdates/EXUpdatesUtils.m

```diff
@@ -118,6 +118,27 @@ + (NSString *)pathForBundledAsset:(EXUpdatesAsset *)asset
     : [[NSBundle mainBundle] pathForResource:asset.mainBundleFilename ofType:asset.type];
 }
 
+// Recreate a UIViewController from old `window.rootViewController`
++ (UIViewController *)createRootViewController:(UIWindow *)window
+{
+  UIViewController* result;
+  UIViewController* oldRootViewController = window.rootViewController;
+
+  SEL screenOrientationSelector = NSSelectorFromString(@"initWithDefaultScreenOrientationMask:");
+  // Support expo-screen-orientation's designated initializer
+  if ([oldRootViewController respondsToSelector:screenOrientationSelector]) {
+    NSNumber *maskValue = (NSNumber *) [oldRootViewController valueForKey:@"defaultOrientationMask"];
+    UIInterfaceOrientationMask mask = (UIInterfaceOrientationMask) [maskValue intValue];
+    id instance = [oldRootViewController.class alloc];
+    IMP imp = [instance methodForSelector:screenOrientationSelector];
+    result = ((id (*)(id, SEL, UIInterfaceOrientationMask))imp)(instance, screenOrientationSelector, mask);
```

Comment:
Can't we just make a separate protocol with this initializer, check if the instance conforms to the protocol and call it in a standard way? 😅

Reply:
> Can't we just make a separate protocol with this initializer, check if the instance conforms to the protocol and call it in a standard way? 😅

yes, that is what i mean for the "interfaces of expo-modules-core". the protocol should be placed in expo-modules-core because it's crossing expo-updates and expo-screen-orientation. if that make sense to you, i can update the pr with the change.

Reply:
Yeah, that's good idea but you can also make it just in expo-updates if you don't want to clutter modules core. In ObjC, the screen orientation class doesn't have to explicitly inherit from that protocol, if you use define the protocol in updates and check `conformsToProtocol` on the object, it will work.

Reply:
@tsapeta looks like it doesn't work. or if i misunderstood something.

```objc
@interface ClassA : NSObject

- (instancetype)initCommon;

@end

@implementation ClassA

- (instancetype)initCommon
{
  return (self = [super init]);
}

@end

@protocol CommonProtocol

- (instancetype)initCommon;

@end

@interface ClassB : NSObject

- (instancetype)initCommon;

@end

@implementation ClassB

- (instancetype)initCommon
{
  return (self = [super init]);
}

@end

@implementation AppDelegate


- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  ClassB *instance = [[ClassB alloc] init];
  if ([instance conformsToProtocol:@protocol(CommonProtocol)]) {
    NSLog(@"conformsToProtocol: YES");
  } else {
    NSLog(@"conformsToProtocol: NO");
  }
  // Override point for customization after application launch.
  return YES;
}
```

the output is : `conformsToProtocol: NO`

if `@interface ClassB : NSObject<CommonProtocol>` then the output is `conformsToProtocol: YES`

Reply:
Yeah, but what if you declare the conformance by the category? Like
```objc
@interface ClassA (CommonCategory) <CommonProtocol>
@end
```

Reply:
Ah, I see — it would have to be for `UIViewController` 🤔 so maybe let's leave it as you did 😉

Reply:
yep, i think so either. adding both @interface and @protocol inside expo-updates for expo-screen-orientation sounds to increase much complexity. let's go with my original approach. thank you, @tsapeta.